src/crimson/os/seastore/extent_placement_manager.cc

```diff
@@ -486,4 +486,58 @@ void ExtentPlacementManager::BackgroundProcess::register_metrics()
   });
 }
 
+RandomBlockOolWriter::alloc_write_iertr::future<>
+RandomBlockOolWriter::alloc_write_ool_extents(
+  Transaction& t,
+  std::list<LogicalCachedExtentRef>& extents)
+{
+  if (extents.empty()) {
+    return alloc_write_iertr::now();
+  }
+  return seastar::with_gate(write_guard, [this, &t, &extents] {
+    return do_write(t, extents);
+  });
+}
+
+RandomBlockOolWriter::alloc_write_iertr::future<>
+RandomBlockOolWriter::do_write(
+  Transaction& t,
+  std::list<LogicalCachedExtentRef>& extents)
+{
+  LOG_PREFIX(RandomBlockOolWriter::do_write);
+  assert(!extents.empty());
+  return trans_intr::do_for_each(extents,
+    [this, &t, FNAME](auto& ex) {
+    DEBUG("try to allocate extent size {}", ex->get_length());
+    return rbm->alloc_extent(t, ex->get_length()
```

Comment:
I think doing asynchronous allocation is not the main purpose of RandomBlock.

EPM has reserved the extension point for non-delay allocation path that is designed to return a paddr synchronously. See https://github.com/ceph/ceph/blob/c2adf33ed5a5cb41ed214781f3adc49b01c295b0/src/crimson/os/seastore/extent_placement_manager.h#L136-L140

Probably `RBCleaner` can introduce a `ExtentAllocator` for Gen-0 DATA and Gen-1(dirty) METADATA extents to do allocations there. And `RandomBlockOolWriter` only needs to write down the blocks according to the pre-determined paddr, for the specific generations.

Without evection implemented, Gen-2 DATA/METADATA should be impossible.